pkg/resource/obtenant_task.go

```diff
@@ -0,0 +1,1044 @@
+package resource
+
+import (
+	"fmt"
+	"github.com/oceanbase/ob-operator/api/v1alpha1"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase/const/config"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase/const/status/tenant"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase/model"
+	"github.com/pkg/errors"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ---------- task entry point ----------
+
+func (m *OBTenantManager) CreateTenantTaskWithClear() error {
+	err := m.CreateTenantTask()
+	// clean created resource, restore to the initial state
+	if err != nil {
+		err := m.DeleteTenantTask()
+		if err != nil {
+			err = errors.Wrapf(err,"delete tenant when creating tenant")
+			return err
+		}
+	}
+	return err
+}
+
+func (m *OBTenantManager) CreateTenantTask() error {
+	m.Logger.Info("===== debug: create tenant task =====")
+	tenantName := m.OBTenant.Spec.TenantName
+
+
+	m.Logger.Info("===== debug: Create Tenant begin =====")
+	for _, pool := range m.OBTenant.Spec.Pools {
+		err := m.CreateUnitAndPoolV4(pool)
```

Comment:
it's better to seperate create unit, create resource pool, create tenant into different tasks

Reply:
Unit and pool have not been split yet because they are one-to-one and somewhat coupled in implementation.
We can decouple them in future commit.